src/main/webapp/assets/js/signUpScript.js

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file provides the JavaScript induced on the sign up page (sign-up.html) which is
+// a replica from the main page (index.html) without additional functionality to load posts.
+
+//
+// Event listener registration
+//
+
+// Hooks the onLoad function to the DOMContentLoaded event.
+document.addEventListener('DOMContentLoaded', onLoad);
+
+//
+// Functions
+//
+
+/**
+ * Fires as soon as the DOM is loaded.
+ */
+async function onLoad() {
+  const collegeLocations = await (await fetch('./assets/college-locations.json')).json();
+
+  // Grab the datalist and remove its ID (destroying the select-datalist relationship),
+  // to improve performance while adding the options to the datalist.
+  const collegeDataList = document.getElementById('colleges');
+  collegeDataList.removeAttribute('id');
+
+  // Add all colleges as datalist options. We use document fragments because the
+  // DOM is slow if we add each option individually and let the DOM update in between.
```

Comment:
These are very good comments, explaining the `why`.

Reply:
Thank you! 😁